pkg/modelfinder/modelfinder.go

```diff
@@ -0,0 +1,169 @@
+package modelfinder
+
+import (
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+var prefixListToIgnore = []string{".cog", ".git", "__pycache__"}
+
+var suffixListToIgnore = []string{
+	".py", ".ipynb", ".whl", // python projects
+	".jpg", ".jpeg", ".png", ".webp", ".svg", ".gif", ".mp4", // images and videos
+	".log", // logs
+}
+
+// ModelFinder is an interface that finds all the model files in the current directory.
+type ModelFinder interface {
+	// FindModels finds all the model files in the current directory.
+	// It returns a list of directories (except current directory) that contain model files,
+	// and a list of model files in current directory.
+	FindModels() ([]string, []string, error)
+}
+
+type FileWalker func(root string, walkFn filepath.WalkFunc) error
+
+type modelFinder struct {
+	fw FileWalker
+}
+
+func New() ModelFinder {
```

Comment:
What do you think about rewriting this as a top-level function instead of defining a new type? Any reason `FindModels` couldn't take a `FileWalker` argument directly?